build fails on clang

```
 ~/r/oxce   oxce-plus $…  ninja -C build -j 10 -k 0                                                                                                                                                     2367ms  2023년 12월 10일 (일) 오전 07시 40분 57초ninja: Entering directory `build'
[1/3] Building CXX object src/CMakeFiles/openxcom.dir/Battlescape/Inventory.cpp.o
FAILED: src/CMakeFiles/openxcom.dir/Battlescape/Inventory.cpp.o 
/usr/bin/ccache /usr/bin/clang++  -I/usr/include/SDL -I/home/scarf/repo/oxce/build -std=c++17 -O3 -DNDEBUG -std=gnu++17 -Wall -Wextra -Winit-self -Wmissing-include-dirs -Wunknown-pragmas -Wmissing-format-attribute -Wredundant-decls -Wformat-security -Wtype-limits -Wempty-body -Wuninitialized -Wignored-qualifiers -Wno-unused-parameter -Wno-inline -Qunused-arguments -pipe -MD -MT src/CMakeFiles/openxcom.dir/Battlescape/Inventory.cpp.o -MF src/CMakeFiles/openxcom.dir/Battlescape/Inventory.cpp.o.d -o src/CMakeFiles/openxcom.dir/Battlescape/Inventory.cpp.o -c /home/scarf/repo/oxce/src/Battlescape/Inventory.cpp
/home/scarf/repo/oxce/src/Battlescape/Inventory.cpp:754:40: error: use of undeclared identifier 'INT_MAX'
  754 |                                                                 int cheapestCostToMoveToHand = INT_MAX;
      |                                                                                                ^
1 error generated.
[2/3] Building CXX object src/CMakeFiles/openxcom.dir/Savegame/BattleUnit.cpp.o
FAILED: src/CMakeFiles/openxcom.dir/Savegame/BattleUnit.cpp.o 
/usr/bin/ccache /usr/bin/clang++  -I/usr/include/SDL -I/home/scarf/repo/oxce/build -std=c++17 -O3 -DNDEBUG -std=gnu++17 -Wall -Wextra -Winit-self -Wmissing-include-dirs -Wunknown-pragmas -Wmissing-format-attribute -Wredundant-decls -Wformat-security -Wtype-limits -Wempty-body -Wuninitialized -Wignored-qualifiers -Wno-unused-parameter -Wno-inline -Qunused-arguments -pipe -MD -MT src/CMakeFiles/openxcom.dir/Savegame/BattleUnit.cpp.o -MF src/CMakeFiles/openxcom.dir/Savegame/BattleUnit.cpp.o.d -o src/CMakeFiles/openxcom.dir/Savegame/BattleUnit.cpp.o -c /home/scarf/repo/oxce/src/Savegame/BattleUnit.cpp
/home/scarf/repo/oxce/src/Savegame/BattleUnit.cpp:393:100: warning: initializer order does not match the declaration order [-Wreorder-ctor]
  393 |         _toDirectionTurret(0), _verticalDirection(0), _status(STATUS_STANDING), _wantsToSurrender(false), _hasPanickedLastTurn(false), _isSurrendering(false), _walkPhase(0),
      |                                                                                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~
      |                                                                                                           _isSurrendering(false)       _hasPanickedLastTurn(false)
  394 |         _fallPhase(0), _kneeled(false), _floating(false), _dontReselect(false), _fire(0), _currentAIState(0),
  395 |         _visible(false), _exp{ }, _expTmp{ },
      |         ~~~~~~~~~~~~~~~  ~~~~~~~  ~~~~~~~~~~
      |         _allowAutoCombat(true) _visible(false) _exp{ }
  396 |         _motionPoints(0), _scannedTurn(-1), _kills(0), _hitByFire(false), _hitByAnything(false), _alreadyExploded(false), _fireMaxHit(0), _smokeMaxHit(0),
      |         ~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~  ~~~~~~~~~  ~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~
      |         _expTmp{ }        _motionPoints(0)  _scannedTurn(-1) _kills(0)    _hitByFire(false)      _hitByAnything(false)    _alreadyExploded(false) _fireMaxHit(0)
  397 |         _moraleRestored(0), _charging(0), _turnsSinceSpotted(255), _turnsLeftSpottedForSnipers(0),
      |         ~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         _smokeMaxHit(0)     _moraleRestored(0) _charging(0)        _turnsSinceSpotted(255)
  398 |         _turnsSinceSeenByHostile(255), _turnsSinceSeenByNeutral(255), _turnsSinceSeenByPlayer(255),
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         _turnsLeftSpottedForSnipers(0) _turnsSinceSeenByHostile(255)  _turnsSinceSeenByNeutral(255)
  399 |         _tileLastSpottedByHostile(-1), _tileLastSpottedByNeutral(-1), _tileLastSpottedByPlayer(-1), _tileLastSpottedForBlindShotByHostile(-1), _tileLastSpottedForBlindShotByNeutral(-1), _tileLastSpottedForBlindShotByPlayer(-1),
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         _turnsSinceSeenByPlayer(255)   _tileLastSpottedByHostile(-1)  _tileLastSpottedByNeutral(-1) _tileLastSpottedByPlayer(-1)               _tileLastSpottedForBlindShotByHostile(-1)  _tileLastSpottedForBlindShotByNeutral(-1)
  400 |         _statistics(), _murdererId(0), _mindControllerID(0), _fatalShotSide(SIDE_FRONT),
      |         ~~~~~~~~~~~~~  ~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         _tileLastSpottedForBlindShotByPlayer(-1) _statistics() _murdererId(0) _mindControllerID(0)
  401 |         _fatalShotBodyPart(BODYPART_HEAD), _armor(armor), _geoscapeSoldier(0),  _unitRules(unit),
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~   ~~~~~~~~~~~~~~~~
      |         _fatalShotSide(SIDE_FRONT)         _fatalShotBodyPart(BODYPART_HEAD) _armor(armor) _geoscapeSoldier(0)
  402 |         _rankInt(0), _turretType(-1), _hidingForTurn(false), _respawn(false), _alreadyRespawned(false),
      |         ~~~~~~~~~~~  ~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~
      |         _unitRules(unit) _rankInt(0)  _turretType(-1)        _hidingForTurn(false) _respawn(false)
  403 |         _isLeeroyJenkins(false), _summonedPlayerUnit(false), _resummonedFakeCivilian(false), _pickUpWeaponsMoreActively(false), _disableIndicators(false),
      |         ~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~
      |         _alreadyRespawned(false) _isLeeroyJenkins(false)     _summonedPlayerUnit(false)      _resummonedFakeCivilian(false)     _pickUpWeaponsMoreActively(false)
  404 |         _vip(false), _bannedInNextStage(false), _allowAutoCombat(true)
      |         ~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~
      |         _disableIndicators(false) _vip(false)   _bannedInNextStage(false)
/home/scarf/repo/oxce/src/Savegame/BattleUnit.cpp:393:100: note: field '_hasPanickedLastTurn' will be initialized after field '_isSurrendering'
  393 |         _toDirectionTurret(0), _verticalDirection(0), _status(STATUS_STANDING), _wantsToSurrender(false), _hasPanickedLastTurn(false), _isSurrendering(false), _walkPhase(0),
      |                                                                                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/scarf/repo/oxce/src/Savegame/BattleUnit.cpp:404:15: note: field '_bannedInNextStage' will be initialized after field '_allowAutoCombat'
  404 |         _vip(false), _bannedInNextStage(false), _allowAutoCombat(true)
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/scarf/repo/oxce/src/Savegame/BattleUnit.cpp:3293:37: error: use of undeclared identifier 'INT_MAX'
 3293 |                                         int cheapestCostToMoveToHand = INT_MAX;
      |                                                                        ^
/home/scarf/repo/oxce/src/Savegame/BattleUnit.cpp:5871:9: warning: unused variable 'haveTUtoAttack' [-Wunused-variable]
 5871 |                         bool haveTUtoAttack = true;
      |                              ^~~~~~~~~~~~~~
/home/scarf/repo/oxce/src/Savegame/BattleUnit.cpp:5873:9: warning: unused variable 'haveTUtoAttack' [-Wunused-variable]
 5873 |                         bool haveTUtoAttack = true;
      |                              ^~~~~~~~~~~~~~
/home/scarf/repo/oxce/src/Savegame/BattleUnit.cpp:5875:9: warning: unused variable 'haveTUtoAttack' [-Wunused-variable]
 5875 |                         bool haveTUtoAttack = true;
      |                              ^~~~~~~~~~~~~~
/home/scarf/repo/oxce/src/Savegame/BattleUnit.cpp:5877:9: warning: unused variable 'haveTUtoAttack' [-Wunused-variable]
 5877 |                         bool haveTUtoAttack = true;
      |                              ^~~~~~~~~~~~~~
/home/scarf/repo/oxce/src/Savegame/BattleUnit.cpp:5879:9: warning: unused variable 'haveTUtoAttack' [-Wunused-variable]
 5879 |                         bool haveTUtoAttack = true;
      |                              ^~~~~~~~~~~~~~
6 warnings and 1 error generated.
ninja: build stopped: cannot make progress due to previous errors.
```

--- a/src/Battlescape/Inventory.cpp
+++ b/src/Battlescape/Inventory.cpp
@@ -19,6 +19,7 @@
 #include "Inventory.h"
 #include <algorithm>
 #include <cmath>
+#include <climits>
 #include "../Mod/Mod.h"
 #include "../Mod/RuleInventory.h"
 #include "../Mod/RuleInterface.h"

--- a/src/Savegame/BattleUnit.cpp
+++ b/src/Savegame/BattleUnit.cpp
@@ -391,18 +391,18 @@ void BattleUnit::prepareBannedFlag(const RuleStartingCondition* sc)
 BattleUnit::BattleUnit(const Mod *mod, Unit *unit, UnitFaction faction, int id, const RuleEnviroEffects* enviro, Armor *armor, StatAdjustment *adjustment, int depth, const RuleStartingCondition* sc) :
 	_faction(faction), _originalFaction(faction), _killedBy(faction), _id(id),
 	_tile(0), _lastPos(Position()), _direction(0), _toDirection(0), _directionTurret(0),
-	_toDirectionTurret(0), _verticalDirection(0), _status(STATUS_STANDING), _wantsToSurrender(false), _hasPanickedLastTurn(false), _isSurrendering(false), _walkPhase(0),
+	_toDirectionTurret(0), _verticalDirection(0), _status(STATUS_STANDING), _wantsToSurrender(false), _isSurrendering(false), _hasPanickedLastTurn(false), _walkPhase(0),
 	_fallPhase(0), _kneeled(false), _floating(false), _dontReselect(false), _fire(0), _currentAIState(0),
-	_visible(false), _exp{ }, _expTmp{ },
-	_motionPoints(0), _scannedTurn(-1), _kills(0), _hitByFire(false), _hitByAnything(false), _alreadyExploded(false), _fireMaxHit(0), _smokeMaxHit(0),
-	_moraleRestored(0), _charging(0), _turnsSinceSpotted(255), _turnsLeftSpottedForSnipers(0),
-	_turnsSinceSeenByHostile(255), _turnsSinceSeenByNeutral(255), _turnsSinceSeenByPlayer(255),
-	_tileLastSpottedByHostile(-1), _tileLastSpottedByNeutral(-1), _tileLastSpottedByPlayer(-1), _tileLastSpottedForBlindShotByHostile(-1), _tileLastSpottedForBlindShotByNeutral(-1), _tileLastSpottedForBlindShotByPlayer(-1),
-	_statistics(), _murdererId(0), _mindControllerID(0), _fatalShotSide(SIDE_FRONT),
-	_fatalShotBodyPart(BODYPART_HEAD), _armor(armor), _geoscapeSoldier(0),  _unitRules(unit),
-	_rankInt(0), _turretType(-1), _hidingForTurn(false), _respawn(false), _alreadyRespawned(false),
-	_isLeeroyJenkins(false), _summonedPlayerUnit(false), _resummonedFakeCivilian(false), _pickUpWeaponsMoreActively(false), _disableIndicators(false),
-	_vip(false), _bannedInNextStage(false), _allowAutoCombat(true)
+	_allowAutoCombat(true), _visible(false), _exp{ },
+	_expTmp{ }, _motionPoints(0), _scannedTurn(-1), _kills(0), _hitByFire(false), _hitByAnything(false), _alreadyExploded(false), _fireMaxHit(0),
+	_smokeMaxHit(0), _moraleRestored(0), _charging(0), _turnsSinceSpotted(255),
+	_turnsLeftSpottedForSnipers(0), _turnsSinceSeenByHostile(255), _turnsSinceSeenByNeutral(255),
+	_turnsSinceSeenByPlayer(255), _tileLastSpottedByHostile(-1), _tileLastSpottedByNeutral(-1), _tileLastSpottedByPlayer(-1), _tileLastSpottedForBlindShotByHostile(-1), _tileLastSpottedForBlindShotByNeutral(-1),
+	_tileLastSpottedForBlindShotByPlayer(-1), _statistics(), _murdererId(0), _mindControllerID(0),
+	_fatalShotSide(SIDE_FRONT), _fatalShotBodyPart(BODYPART_HEAD), _armor(armor),  _geoscapeSoldier(0),
+	_unitRules(unit), _rankInt(0), _turretType(-1), _hidingForTurn(false), _respawn(false),
+	_alreadyRespawned(false), _isLeeroyJenkins(false), _summonedPlayerUnit(false), _resummonedFakeCivilian(false), _pickUpWeaponsMoreActively(false),
+	_disableIndicators(false), _vip(false), _bannedInNextStage(false)
 {
 	if (enviro)
 	{

--- a/src/Savegame/BattleUnit.cpp
+++ b/src/Savegame/BattleUnit.cpp
@@ -20,6 +20,7 @@
 #include "BattleItem.h"
 #include <sstream>
 #include <algorithm>
+#include <climits>
 #include "../Engine/Collections.h"
 #include "../Engine/Surface.h"
 #include "../Engine/Script.h"
@@ -113,7 +114,7 @@ BattleUnit::BattleUnit(const Mod *mod, Soldier *soldier, int depth, const RuleSt
 	_statistics = new BattleUnitStatistics();
 
 	deriveRank();
-	
+
 	_allowAutoCombat = soldier->getAllowAutoCombat();
 
 	updateArmorFromSoldier(mod, soldier, soldier->getArmor(), depth, false, sc);

